app/views/shared/registrations/_ceased_revoked_panel.html.erb

```diff
@@ -0,0 +1,16 @@
+<% if resource.revoked? || resource.inactive? %>
```

Comment:
I'm going start sounding like a stuck record soon 😳 !

Do you think some of the logic in this partial are candidates for being moved to the `RegistrationPresenter`?

Specifically, I'm thinking of

```ruby
<% if resource.revoked? || resource.inactive? %>
```
```ruby
      <% if resource.revoked? %>
        <%= t(".heading.revoked") %>
      <% elsif resource.inactive? %>
        <%= t(".heading.ceased") %>
      <% end %>
```
```ruby
<% if resource.metaData.revoked_reason.present? && can?(:view_revoked_reasons, resource) %>
```

Reply:
I actually started putting some of them in the presenter but it seemed like I was just creating more code for things that were fairly simple and readable as-is. Maybe we need to agree a principle of when things go in the presenter and when they don't?

Being able to check abilities in the presenter will also require us to make some changes to the ability model – nothing major but it is an extra thing.

Reply:
Don't worry about the last one, but I would push for the others.

IMHO moving things to a presenter

- makes them easier to test
- makes them _more_ readable ('cause we lose the ERB cruft)
- makes the view clearer as its more about the structure and less about code

My general principle is if its anything more than an `if foo do bar` I'd start thinking about a presenter. But you are right, we need to agree a team principle for this. Who wants ☕️ and 🍰 later!?

Reply:
@Cruikshanks Sounds good to me! 👍